equinix/config.go

```diff
@@ -128,11 +132,30 @@ func (c *Config) Load(ctx context.Context) error {
 	neClient.SetHeaders(map[string]string{
 		"User-agent": c.fullUserAgent("equinix/ne-go"),
 	})
+	authConfig := oauth2.Config{}
+	if c.Token == "" {
+		authConfig = oauth2.Config{
```

Comment:
It seems logic here can be simplified, actually authConfig is already initialized before, but is not a big deal now and can be refactor later.

https://github.com/equinix/terraform-provider-equinix/pull/209/files#diff-5d56dbc70ed6632fd0523ea2cb89e3afc4a82621b68c6d4010cabfd5cafb2304R113-R117